packages/table/test/harness.ts

```diff
@@ -13,6 +13,80 @@ import * as ReactDOM from "react-dom";
 export type MouseEventType = "click" | "mousedown" | "mouseup" | "mousemove" | "mouseenter" | "mouseleave" ;
 export type KeyboardEventType = "keypress" | "keydown" |  "keyup" ;
 
+function dispatchTestKeyboardEvent(target: EventTarget, eventType: string, key: string, modKey = false) {
+    const event = document.createEvent("KeyboardEvent");
+    const keyCode = key.charCodeAt(0);
+
+    let ctrlKey = false;
+    let metaKey = false;
+
+    if (modKey) {
+        if ((typeof navigator !== "undefined") && /Mac|iPod|iPhone|iPad/.test(navigator.platform)) {
+            metaKey = true;
+        } else {
+            ctrlKey = true;
+        }
+    }
+
+    (event as any).initKeyboardEvent(eventType, true, true, window, key, 0, ctrlKey, false, false, metaKey);
+
+    // Hack around these readonly properties in WebKit and Chrome
+    if (detectBrowser() === Browser.WEBKIT) {
+        (event as any).key = key;
+        (event as any).which = keyCode;
+    } else {
+        Object.defineProperty(event, "key", { get: () => key });
+        Object.defineProperty(event, "which", { get: () => keyCode });
+    }
+
+    target.dispatchEvent(event);
+}
+
+/**
+ * Enum of possible browsers
+ */
+enum Browser {
+    CHROME,
+    EDGE,
+    FIREFOX,
+    IE,
+    UNKNOWN,
+    WEBKIT,
+}
+
+/**
+ * Use feature detection to determine current browser.
```

Comment:
lol this is kinda backwards... you're supposed to use feature detection _instead of_ browser detection: ensure that the feature you want exists, rather than asserting the browser "make." it's version-independent and much less prone to false positives.

is there a more reliable way of doing these checks that we can shim in here quickly?

at the very least move this to its own file.